diff --color -Naur src_original/bbox.cpp src_patched/bbox.cpp
--- src_original/bbox.cpp	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/bbox.cpp	2024-02-28 17:43:09.810635128 +0000
@@ -49,7 +49,9 @@
 #include "lightgrp.h"
 #include "povmsend.h"
 
+#ifndef ININSIDEPOV
 BEGIN_POV_NAMESPACE
+#endif
 
 /*****************************************************************************
 * Local preprocessor defines
@@ -1893,4 +1895,7 @@
   }
 }
 
+
+#ifndef ININSIDEPOV
 END_POV_NAMESPACE
+#endif
diff --color -Naur src_original/bbox.h src_patched/bbox.h
--- src_original/bbox.h	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/bbox.h	2024-02-28 17:43:09.810635128 +0000
@@ -34,7 +34,10 @@
 #ifndef BBOX_H
 #define BBOX_H
 
-BEGIN_POV_NAMESPACE
+// FLAVIO - MAYBE WILL BEING PROBLEM
+// #ifndef ININSIDEPOV
+// BEGIN_POV_NAMESPACE
+// #endif
 
 /*****************************************************************************
 * Global preprocessor defines
@@ -117,6 +120,9 @@
 	a = b.Lengths[X] * b.Lengths[Y] * b.Lengths[Z];
 }
 
-END_POV_NAMESPACE
+// FLAVIO - MAYBE WILL BEING PROBLEM
+// #ifndef ININSIDEPOV
+// END_POV_NAMESPACE
+// #endif
 
 #endif
diff --color -Naur src_original/boxes.cpp src_patched/boxes.cpp
--- src_original/boxes.cpp	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/boxes.cpp	2024-02-28 17:43:09.810635128 +0000
@@ -39,7 +39,9 @@
 #include "matrices.h"
 #include "objects.h"
 
+#ifndef ININSIDEPOV
 BEGIN_POV_NAMESPACE
+#endif
 
 /*****************************************************************************
 * Local preprocessor defines
@@ -1198,4 +1200,6 @@
   */
 }
 
+#ifndef ININSIDEPOV
 END_POV_NAMESPACE
+#endif
diff --color -Naur src_original/boxes.h src_patched/boxes.h
--- src_original/boxes.h	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/boxes.h	2024-02-28 17:43:09.810635128 +0000
@@ -33,7 +33,9 @@
 #ifndef BOXES_H
 #define BOXES_H
 
+#ifndef ININSIDEPOV
 BEGIN_POV_NAMESPACE
+#endif
 
 /*****************************************************************************
 * Global preprocessor defines
@@ -49,9 +51,9 @@
 
 typedef struct Box_Struct BOX;
 
-struct Box_Struct
+/* OBJECT_FIELDS */
+struct Box_Struct : Object_Struct
 {
-  OBJECT_FIELDS
   VECTOR bounds[2];
 };
 
@@ -75,6 +77,8 @@
 BOX *Copy_Box (OBJECT *Object);
 void Compute_Box_BBox (BOX *Box);
 
+#ifndef ININSIDEPOV
 END_POV_NAMESPACE
+#endif
 
 #endif
diff --color -Naur src_original/cones.h src_patched/cones.h
--- src_original/cones.h	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/cones.h	2024-02-28 17:43:09.810635128 +0000
@@ -50,9 +50,9 @@
 typedef struct Cone_Struct CYLINDER;
 typedef struct Cone_Struct CONE;
 
-struct Cone_Struct
+/* OBJECT_FIELDS */
+struct Cone_Struct : public Object_Struct
 {
-  OBJECT_FIELDS
   VECTOR apex;        /* Center of the top of the cone */
   VECTOR base;        /* Center of the bottom of the cone */
   DBL apex_radius;    /* Radius of the cone at the top */
diff --color -Naur src_original/csg.h src_patched/csg.h
--- src_original/csg.h	2024-02-28 17:43:19.514670742 +0000
+++ src_patched/csg.h	2024-02-28 17:43:09.810635128 +0000
@@ -33,7 +33,9 @@
 #ifndef CSG_H
 #define CSG_H
 
+#ifndef INSIDEPOV
 BEGIN_POV_NAMESPACE
+#endif
 
 /*****************************************************************************
 * Global preprocessor defines
@@ -59,9 +61,9 @@
 
 typedef struct CSG_Struct CSG;
 
-struct CSG_Struct
+/* COMPOUND_FIELDS */
+struct CSG_Struct : public Compound_Object_Struct
 {
-  COMPOUND_FIELDS
   int do_split;
 };
 
@@ -87,6 +89,8 @@
 void Compute_CSG_BBox (OBJECT *Object);
 void Determine_CSG_Textures(CSG *Csg, VECTOR IPoint, int *Count, TEXTURE **Textures, DBL *Weights);
 
+#ifndef INSIDEPOV
 END_POV_NAMESPACE
+#endif
 
 #endif
diff --color -Naur src_original/frame.h src_patched/frame.h
--- src_original/frame.h	2024-02-28 17:43:19.518670756 +0000
+++ src_patched/frame.h	2024-02-28 17:43:09.810635128 +0000
@@ -696,6 +696,8 @@
 typedef COLC RGB[3];
 typedef SNGL SNGL_VECT[3];
 
+
+
 /* Vector array elements. */
 enum
 {
@@ -1334,20 +1336,20 @@
    OBJECT *Object;\
  } Vals;
 
-struct Pattern_Struct
+struct Pattern_Struct 
 {
   TPATTERN_FIELDS
 };
 
-struct Pigment_Struct
+/* TPATTERN_FIELDS */
+struct Pigment_Struct : public Pattern_Struct
 {
-  TPATTERN_FIELDS
   COLOUR Colour; 
 };
 
-struct Tnormal_Struct
+/* TPATTERN_FIELDS */
+struct Tnormal_Struct : public Pattern_Struct
 {
-  TPATTERN_FIELDS
   SNGL Amount;
   SNGL Delta; /* NK delta */
 };
@@ -1356,9 +1358,12 @@
   TPATTERN_FIELDS      \
   TEXTURE *Next_Material;
 
-struct Texture_Struct
+
+/* TEXTURE_FIELDS */
+
+struct Texture_Struct : public Pattern_Struct
 {
-  TEXTURE_FIELDS
+  TEXTURE *Next_Material; // flavio
   PIGMENT *Pigment;
   TNORMAL *Tnormal;
   FINISH *Finish;
@@ -1389,9 +1394,9 @@
   WARP_FIELDS
 };
 
-struct Turb_Struct
+/* WARP_FIELDS */
+struct Turb_Struct : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR Turbulence;
   int Octaves;
   SNGL Lambda, Omega;
@@ -1537,12 +1542,17 @@
   OBJECT_FIELDS
 };
 
-struct Compound_Object_Struct
+/* COMPOUND_FIELDS */
+struct Compound_Object_Struct : public Object_Struct
 {
-  COMPOUND_FIELDS
+  OBJECT *Children;
 };
 
 
+// FLAVIO NEW INCLUDES
+#define INSIDEPOV 1
+#include "csg.h"
+
 typedef struct BBox_Tree_Struct BBOX_TREE;
 
 struct BBox_Tree_Struct
@@ -1561,6 +1571,9 @@
   int x1, y1, x2, y2;
 };
 
+
+#include "vlbuffer.h"
+
 /*
  * The following structure represent the bounding box hierarchy in 2d space.
  * Because is_leaf, Object and Project are the first elements in both
@@ -1568,18 +1581,20 @@
  * a pointer is pointing.
  */
 
-struct Project_Tree_Node_Struct
-{
-  unsigned short is_leaf;
+/* unsigned short is_leaf; 
   BBOX_TREE *Node;
   PROJECT Project;
+ */
+struct Project_Tree_Node_Struct : public Project_Tree_Leaf_Struct
+{
   unsigned short Entries;
   PROJECT_TREE_NODE **Entry;
 };
 
-struct Light_Source_Struct
+/* COMPOUND_FIELDS */
+struct Light_Source_Struct : public CSG_Struct //  Compound_Object_Struct
 {
-  COMPOUND_FIELDS
+  
   COLOUR Colour;
   VECTOR Direction, Center, Points_At, Axis1, Axis2;
   DBL Coeff, Radius, Falloff;
@@ -1930,3 +1945,4 @@
 #endif
 
 #endif
+
diff --color -Naur src_original/frontend/processrenderoptions.cpp src_patched/frontend/processrenderoptions.cpp
--- src_original/frontend/processrenderoptions.cpp	2024-02-28 17:43:19.518670756 +0000
+++ src_patched/frontend/processrenderoptions.cpp	2024-02-28 17:43:09.814635143 +0000
@@ -338,7 +338,7 @@
 			err = POVMSUtil_SetInt(obj, option->key, tolower(*param));
 			break;
 		case kPOVAttrib_IncludeIni:
-		case kPOVAttrib_LibraryPath:
+		case kPOVAttrib_LibraryPath: {
 			POVMSAttribute attr;
 
 			if(err == kNoErr)
@@ -378,7 +378,8 @@
 			if(err == kNoErr)
 				err = POVMSObject_Set(obj, &list, option->key);
 			break;
-		case kPOVAttrib_Declare:
+		}
+		case kPOVAttrib_Declare: {
 			POVMSObject decobj;
 
 			// create list if it isn't there
@@ -422,6 +423,7 @@
 			if(err == kNoErr)
 				err = POVMSObject_Set(obj, &list, option->key);
 			break;
+		}
 		case kPOVAttrib_FatalErrorCommand:
 		case kPOVAttrib_PostFrameCommand:
 		case kPOVAttrib_PostSceneCommand:
@@ -526,7 +528,7 @@
 				err = POVMSUtil_SetInt(obj, option->key, chr);
 			}
 			break;
-		case kPOVAttrib_LibraryPath:
+		case kPOVAttrib_LibraryPath: {
 			POVMSAttributeList list;
 			POVMSAttribute attr;
 
@@ -560,6 +562,7 @@
 			if(err == kNoErr)
 				err = POVMSObject_Set(obj, &list, option->key);
 			break;
+		}
 		case kPOVAttrib_TestAbortCount:
 			if((*param) == 0)
 				break;
@@ -610,7 +613,7 @@
 			break;
 		case kPOVAttrib_IncludeIni:
 			break;
-		case kPOVAttrib_Declare:
+		case kPOVAttrib_Declare: {
 			POVMSObject decobj;
 
 			err = POVMSObject_Get(obj, &list, option->key);
@@ -647,6 +650,7 @@
 				}
 			}
 			break;
+		}
 		case kPOVAttrib_LibraryPath:
 			err = POVMSObject_Get(obj, &list, option->key);
 			if(err != 0)
diff --color -Naur src_original/planes.h src_patched/planes.h
--- src_original/planes.h	2024-02-28 17:43:19.526670785 +0000
+++ src_patched/planes.h	2024-02-28 17:43:09.818635158 +0000
@@ -49,9 +49,9 @@
 
 typedef struct Plane_Struct PLANE;
 
-struct Plane_Struct
+/* OBJECT_FIELDS */
+struct Plane_Struct : public Object_Struct
 {
-  OBJECT_FIELDS
   VECTOR Normal_Vector;
   DBL    Distance;
 };
diff --color -Naur src_original/quadrics.h src_patched/quadrics.h
--- src_original/quadrics.h	2024-02-28 17:43:19.526670785 +0000
+++ src_patched/quadrics.h	2024-02-28 17:43:09.822635173 +0000
@@ -51,9 +51,9 @@
 
 typedef struct Quadric_Struct QUADRIC;
 
-struct Quadric_Struct
+/* OBJECT_FIELDS */
+struct Quadric_Struct : public Object_Struct
 {
-  OBJECT_FIELDS
   VECTOR Square_Terms;
   VECTOR Mixed_Terms;
   VECTOR Terms;
diff --color -Naur src_original/spheres.h src_patched/spheres.h
--- src_original/spheres.h	2024-02-28 17:43:19.530670800 +0000
+++ src_patched/spheres.h	2024-02-28 17:43:09.822635173 +0000
@@ -48,9 +48,9 @@
 
 typedef struct Sphere_Struct SPHERE;
 
-struct Sphere_Struct
+/* OBJECT_FIELDS */
+struct Sphere_Struct : public Object_Struct
 {
-  OBJECT_FIELDS
   VECTOR Center;
   DBL    Radius;
 };
diff --color -Naur src_original/vlbuffer.h src_patched/vlbuffer.h
--- src_original/vlbuffer.h	2024-02-28 17:43:19.530670800 +0000
+++ src_patched/vlbuffer.h	2024-02-28 17:43:09.826635188 +0000
@@ -36,7 +36,9 @@
 #include "frame.h"
 #include "bbox.h"
 
+#ifndef INSIDEPOV
 BEGIN_POV_NAMESPACE
+#endif
 
 /*****************************************************************************
 * Global preprocessor defines
@@ -115,6 +117,8 @@
 
 void Destroy_Project_Tree (PROJECT_TREE_NODE *Node);
 
+#ifndef INSIDEPOV
 END_POV_NAMESPACE
+#endif
 
 #endif
diff --color -Naur src_original/warps.h src_patched/warps.h
--- src_original/warps.h	2024-02-28 17:43:19.530670800 +0000
+++ src_patched/warps.h	2024-02-28 17:43:09.826635188 +0000
@@ -66,52 +66,52 @@
 typedef struct Toroidal_Warp TOROIDAL;
 typedef struct Planar_Warp PLANARW;
 
-struct Toroidal_Warp
+/* WARP_FIELDS */
+struct Toroidal_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR Orientation_Vector;
   DBL DistExp;
   DBL MajorRadius;
 };
 
-struct Cylindrical_Warp
+/* WARP_FIELDS */
+struct Cylindrical_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR Orientation_Vector;
   DBL DistExp;
 };
 
-struct Planar_Warp
+/* WARP_FIELDS */
+struct Planar_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR Orientation_Vector;
   DBL OffSet;
 };
 
-struct Spherical_Warp
+/* WARP_FIELDS */
+struct Spherical_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR Orientation_Vector;
   DBL DistExp;
 };
 
-struct Repeat_Warp
+/* WARP_FIELDS */
+struct Repeat_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   int Axis;
   SNGL Width;
   VECTOR Flip, Offset;
 };
 
-struct Trans_Warp
+/* WARP_FIELDS */
+struct Trans_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   TRANSFORM Trans;
 };
 
-struct Black_Hole_Warp
+/* WARP_FIELDS */
+struct Black_Hole_Warp : public Warps_Struct
 {
-  WARP_FIELDS
   VECTOR      Center ;
   VECTOR      Repeat_Vector ;
   VECTOR      Uncertainty_Vector ;
